### PR TITLE
Enable search by user handle

### PR DESCRIPTION
**Pull request by vytis**
It seems that local search by user handle was not working, it was searching only by `normalizedName`. Updated predicates to handle this situation and re-using it in team members search.

### CODE REVIEW

File: Source/Model/User/ZMUser+Internal.h [hunk 1]
```diff
@@ -57,23 +57,6 @@ extern NSString * __nonnull const ZMUserActiveConversationsKey;
...
-
-/// @method predicateForConnectedNonBotUsers
-/// Retrieves all users (excluding bots), having ZMConnectionStatusAccepted connection statuses.
-+ (nonnull NSPredicate *)predicateForConnectedNonBotUsers;
-
-/// @method predicateForUsersWithSearchString:connectionStatusInArray:
-/// Retrieves users with name or email matching search string, having one of given connection statuses.
-/// @param searchString - a predicate to search users
-/// @param connectionStatusArray - an array of connections status of the users. E.g. for connected users it is @[@(ZMConnectionStatusAccepted)]
-+ (nonnull NSPredicate *)predicateForUsersWithSearchString:(nonnull NSString *)searchString
```
Comment (typfel) [inline Source/Model/User/ZMUser+Internal.h]: Shouldn't the `*.m` implementation be deleted as well?

Comment (vytis) [inline Source/Model/User/ZMUser+Internal.h]: 🤦‍♂️  good catch!

File: Source/Model/User/ZMUser.swift [hunk 2]
```diff
@@ -78,6 +79,83 @@ extension ZMUser {
...
+                var withoutAt = query
+                withoutAt.remove(at: query.startIndex)
+                normalizedHandle = withoutAt
+            } else {
+                // User regular normalized query otherwise
+                normalizedHandle = normalizedQuery
+            }
+            
+            let handlePredicate = NSPredicate(format: "%K BEGINSWITH %@", #keyPath(ZMUser.handle), normalizedHandle)
+            allPredicates.append([namePredicate, handlePredicate].flatMap {$0})
```
Comment (mikeger) [inline Source/Model/User/ZMUser.swift]: Why `flatMap` is necessary? I think that they are not optional.

Comment (vytis) [inline Source/Model/User/ZMUser.swift]: The `handlePredicate` is force-unwrapped from the obj-c land. Was thinking to use regular predicate but this seems to have lots of search-smartness inside and actually sprints query into words and matches them separately.

File: Source/Model/User/ZMUser+Internal.h (hunk 1 above)
Comment (typfel) [inline Source/Model/User/ZMUser+Internal.h]: Were there also tests for the old implementation which can be deleted?

Comment (vytis) [inline Source/Model/User/ZMUser+Internal.h]: Doesn't seem like there were any tests for the predicates directly, only for `ZMSearchDirectory` in SE.

Comment (vytis) [inline Source/Model/User/ZMUser+Internal.h]: Maybe we should start using some sort of code coverage tool to catch these sort of cases...

**Comment by mikeger** [comment]
@vytis I forgot to mention, let's not merge new stuff yet until we have our RC done, otherwise we are risking not to end it ever :)

**Pull request merged**